code/lib/manager-api/src/index.tsx

```diff
@@ -416,7 +416,7 @@ const addonStateCache: {
 } = {};
 
 // shared state
-export function useSharedState<S>(stateId: string, defaultState?: S) {
+export function useSharedState<S extends object>(stateId: string, defaultState?: S) {
```

Comment:
`S extends object`

- doesn't everything extend object in JS land?
- AFAIK `S` could also be a `string`, `number`, etc.

Reply:
Oh I didn't know it could be primitives like `string` or `number`.

I was merely trying to filter out `API_StateMerger<S>` from `S | API_StateMerger<S>` so I can call it, but TypeScript produces:
```
This expression is not callable.
  Not all constituents of type 'API_StateMerger<S> | (S & Function)' are callable.
    Type 'S & Function' has no call signatures.
```

So I added that to work around the typescript error.

What would be a better idea? should I exhaustively include all possible type for `S`?

Reply:
@ndelangen Maybe this is better? https://github.com/storybookjs/storybook/pull/23804/commits/63f2dbef61ad47161d053617a0087cd05deb4766